src/content-tags-drawer/ContentTagsCollapsible.jsx

```diff
@@ -98,99 +239,138 @@ import useContentTagsCollapsibleHelper from './ContentTagsCollapsibleHelper';
+    // Get tags that were unstaged to remove them from checkbox selector
+    const unstagedTags = stagedContentTags.filter(
+      t1 => !stagedTags.some(t2 => t1.value === t2.value),
+    );
+
+    // Call the `tagChangeHandler` with the unstaged tags to unselect them from the selectbox
+    // and update the staged content tags tree. Since the `handleStagedTagsMenuChange` function is={}
+    // only called when a change occurs in the react-select menu component we know that tags can only be
+    // removed from there, hence the tagChangeHandler is always called with `checked=false`.
+    unstagedTags.forEach(unstagedTag => tagChangeHandler(unstagedTag.value, false));
+    setStagedTags(taxonomyId, stagedTags);
+  }, [taxonomyId, setStagedTags, stagedContentTags, tagChangeHandler]);
+
+  const handleCommitStagedTags = React.useCallback(() => {
+    commitStagedTags();
+    handleStagedTagsMenuChange([]);
+    selectRef.current?.blur();
     setSearchTerm('');
-  }, []);
+  }, [commitStagedTags, handleStagedTagsMenuChange, selectRef, setSearchTerm]);
+
+  const handleCancelStagedTags = React.useCallback(() => {
+    handleStagedTagsMenuChange([]);
+    selectRef.current?.blur();
+    setSearchTerm('');
+  }, [handleStagedTagsMenuChange, selectRef, setSearchTerm]);
 
   return (
     <div className="d-flex">
       <Collapsible title={name} styling="card-lg" className="taxonomy-tags-collapsible">
-        <div key={id}>
-          <ContentTagsTree tagsTree={tagsTree} removeTagHandler={tagChangeHandler} />
+        <div key={taxonomyId}>
+          <ContentTagsTree tagsTree={appliedContentTagsTree} removeTagHandler={removeAppliedTagHandler} />
         </div>
 
         <div className="d-flex taxonomy-tags-selector-menu">
 
           {canTagObject && (
-            <Button
-              ref={setAddTagsButtonRef}
-              variant="outline-primary"
-              onClick={open}
-            >
-              <FormattedMessage {...messages.addTagsButtonText} />
-            </Button>
+            <Select
+              ref={/** @type {React.RefObject} */(selectRef)}
+              isMulti
+              isLoading={updateTags.isLoading}
+              isDisabled={updateTags.isLoading}
+              name="tags-select"
+              placeholder={intl.formatMessage(messages.collapsibleAddTagsPlaceholderText)}
+              isSearchable
+              className="d-flex flex-column flex-fill"
+              classNamePrefix="react-select-add-tags"
+              onInputChange={handleSearchChange}
+              onChange={handleStagedTagsMenuChange}
+              components={{
+                // @ts-ignore TODO: Properly fix this
+                Menu: CustomMenu,
+                // @ts-ignore TODO: Properly fix this
+                LoadingIndicator: CustomLoadingIndicator,
+                // @ts-ignore TODO: Properly fix this
+                IndicatorsContainer: CustomIndicatorsContainer,
+              }}
```

Comment:
I couldn't get the types to work properly here, any advice on how to properly fix this?

Reply:
@yusuf-musleh  Sorry for the delay, I'm hoping to check out this PR and answer your question soon :)

Reply:
@yusuf-musleh I figure out the types for you. I meant to push them to a new branch just now, but accidentally pushed them to this same branch. Sorry about that. Please review and incorporate or not as you wish.

Reply:
@bradenmacdonald Thanks a lot for fixing the types! It was really tricky getting them to work. There were some more typing issues the linter was complaining about, but I fixed them here: https://github.com/openedx/frontend-app-course-authoring/pull/834/commits/0200cf7e95a09e695ae54eefc171e3b0a42e8bb2

I also found a react warning in the console caused by setting state of a parent component while updating the state in the child. I spent some time looking into it and refactored it here: https://github.com/openedx/frontend-app-course-authoring/pull/834/commits/cdc2269ae2ea58ebf9d89bb671ad889f16d303db seems to be working fine now.

Reply:
@yusuf-musleh Nice work. This is my first time trying to include a `.d.ts` file in an Open edX MFE with the same name as the `.ts` file, and frankly I'm surprised that it works at all. But it seems to solve the problem :)